worker-sys/src/types/socket.rs

```diff
@@ -0,0 +1,29 @@
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen(module = "cloudflare:sockets")]
+extern "C" {
+    #[wasm_bindgen]
+    pub fn connect(address: JsValue, options: JsValue) -> Socket;
+}
+
+#[wasm_bindgen]
+extern "C" {
+    #[wasm_bindgen(extends=js_sys::Object)]
+    #[derive(Debug, Clone, PartialEq, Eq)]
+    pub type Socket;
+
+    #[wasm_bindgen(method)]
+    pub fn close(this: &Socket) -> js_sys::Promise;
+
+    #[wasm_bindgen(method)]
+    pub fn closed(this: &Socket) -> js_sys::Promise;
+
+    #[wasm_bindgen(method, js_name=startTls)]
+    pub fn start_tls(this: &Socket) -> Socket;
+
+    #[wasm_bindgen(method, getter = readable)]
```

Comment:
It should be okay to just specify `getter`, rather than `getter = foo`, as the function name (`readable`) is the same as JavaScript property.

https://rustwasm.github.io/wasm-bindgen/reference/attributes/on-js-imports/getter-and-setter.html